compiler/rustc_mir_transform/src/dataflow_const_prop.rs

```diff
@@ -403,7 +403,12 @@ impl<'a, 'tcx> ConstAnalysis<'a, 'tcx> {
             operand,
             &mut |elem, op| match elem {
                 TrackElem::Field(idx) => self.ecx.project_field(op, idx.as_usize()).ok(),
-                TrackElem::Variant(idx) => self.ecx.project_downcast(op, idx).ok(),
+                TrackElem::Variant(idx) => {
+                    if op.layout.for_variant(&self.ecx, idx).abi.is_uninhabited() {
```

Comment:
This should be handled when creating the tracked places, skipping recursion in case of uninhabited variant.
Do you have a test-case exhibiting this issue?

Reply:
The mir-opt failures from https://github.com/rust-lang/rust/pull/119627#issuecomment-1879047434